### PR TITLE
feat: Resolving UserWarning about llama-index-llms-dashscope

### PR DESCRIPTION
**Pull request by yuzilog**
In Pydantic V2, protected_namespaces is a configuration option used to prevent certain namespace keywords
          (such as model_, etc.) from being used as field names. so we need to disable it here.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x ] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [x] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-dashscope/llama_index/llms/dashscope/base.py [hunk 1]
```diff
@@ -121,6 +122,12 @@ class DashScope(CustomLLM):
         ```
     """
 
+
+    """ In Pydantic V2, protected_namespaces is a configuration option used to prevent certain namespace keywords
+      (such as model_, etc.) from being used as field names. so we need to disable it here.
+    """
+    model_config = ConfigDict(protected_namespaces=())
```
Comment (nerdai) [inline llama-index-integrations/llms/llama-index-llms-dashscope/llama_index/llms/dashscope/base.py:129]: We typically use something like "pydantic_model" as the protected namespace when there is a clash with default "model"

**Review by nerdai — COMMENTED** [review]
Please bump the version of the package in pyproject.toml as well.

File: llama-index-integrations/llms/llama-index-llms-dashscope/llama_index/llms/dashscope/base.py (hunk 1 above)
Comment (yuzilog) [reply llama-index-integrations/llms/llama-index-llms-dashscope/llama_index/llms/dashscope/base.py:129]: The released version has been used, which will cause version incompatibility.

**Review by nerdai — APPROVED** [review]
Thanks @yuzilog

**Pull request merged**